### DISCUSSION TITLE
Support for SOCKS5

### DISCUSSION THREAD

**Discussion by PieterL75** [napalm-automation/napalm#1427] [Ideas]
We do not expose our network devices direct to any server network.
In order to access the network devices, we need to go though a SOCKS5 proxy (or alternatively an ssh in ssh tunnel).

Can NAPALM support socks5 connections ?

Pieter

**Comment by ktbyers** [comment]
The NAPALM drivers that support SSH (Netmiko-based) should be able to support a socks proxy. Netmiko has a `sock=` argument that you can pass in. You would do this in NAPALM via `optional_args`. This would be for the Cisco IOS and Cisco "nxos_ssh" drivers. IOS-XR is probably, but you might need to dig into it a bit more as it uses Netmiko but there is an intermediate library in there that you would need to verify can pass all of the Netmiko arguments.

I am not sure about Juniper and PyEZ. You would need to look into whether PyEZ has a way of supporting this (i.e. passing the NETCONF over SSH via a socks proxy).

You would also need to look into the HTTP/TLS based drivers (Arista eAPI and NX-OS nx-api).
